docs/configuring.md

```diff
@@ -0,0 +1,241 @@
+```
+
+The OpenJ9 VM supports large page usage only to back shared memory segments. Large pages are allocated by using `shmget()` with the **SHM_LGPG** and **SHM_PIN** flags.
+
+For more information, including support considerations, see [Large pages](https://www.ibm.com/support/knowledgecenter/ssw_aix_72/performance/large_page_ovw.html) and [Multiple page size support](https://www.ibm.com/support/knowledgecenter/ssw_aix_72/performance/multiple_page_size_support.html) in the AIX documentation.
+
+The 16 MB and 16 GB page sizes, which are intended for very high performance environments, require special user permissions. You must also configure the number of pages that you require, which cannot be adjusted on demand. For 16 MB large pages, you set the number of large pages by using the `vmo` command. For 64 GB huge pages you must define the number of pages by using the hardware management console. For more information, see [Page sizes for very high-performance environments](https://www.ibm.com/support/knowledgecenter/ssw_aix_72/performance/page_sizes_very_high-perf_env.html) in the AIX documentation.
+
+### Linux systems
+
+Large pages are typically referred to as *huge pages* on Linux systems. To configure huge page memory allocation, the kernel must support huge pages. If huge pages are supported, the following lines are present in the **proc/meminfo** file:
+
+```
+HugePages_Total:    
+HugePages_Free:     
+Hugepagesize:     
+```
+
+If these lines do not exist, update your Linux kernel. Typically, huge pages are enabled by default. However, if `HugePages_Total` has a value of `0`, huge pages are available, but not enabled. To enable huge pages, add the following line to your **/etc/sysctl.conf** file and reload the configuration by running `sysctl -p`:
+
+```
+vm.nr_hugepages=<number>
+```
+
+Where `<number>` is the number of huge pages required.
+
+Configure the number of huge pages that you require at boot time to ensure that the VM has access to sufficient contiguous pages. The following kernel parameters must be set appropriately for your system:
+
+- **SHMMAX**: The maximum size of the shared memory segment (bytes).
+- **SHMALL**: The total amount of shared memory in the system (bytes or pages).
+
+The user running the Java process must either be ROOT or have permissions to use huge pages. For the appropriate permissions, the user must be a member of a group that has its group identifier (gid) stored in **/proc/sys/vm/hugetlb_shm_group**. The locked memory limit must also be increased to at least the size of the Java heap by using the **ulimit -l** command.
+
+To promote all memory allocated to huge pages, set the `-XX:+TransparentHugePage` option on the command line when you start your application and ensure that the Transparent Huge Page option on your system is set to `madvise`.
+
+Where huge page support is available, the following default sizes apply for the object heap:
+
+- Linux on x86: 2 MB
+- Linux on IBM POWER: 16 MB
```

Comment:
starting POWER9 and later, there are two types of huge pages on Linux:  16MB and 2MB.  They are available exclusively (i.e. 16MB page availability doesn't co-exist with 2MB page availability). As currently, PowerVM LPAR(s)/machines will offer 16MB huge pages, while bare-metal machines will offer 2MB huge pages.  In the future, PowerVM environment may support 2MB huge pages (but not now).

Reply:
@zl-wang the doc is updated to state "bare-metal machines support only 2 MB huge pages", is "only" correct?

Reply:
it is correct as of after-a-certain-version of kernel.  such that, as a forward-looking document, it is correct.  if this document meant to describe older kernel situations as well, i would suggest changing the wording on bare-metal case to be "either 16MB or 2MB huge pages, depending on the exact version of the Linux kernel".  Too mouthful maybe. of course, a better more concise description (to the same meaning) would be very welcome.

Reply:
So: 

- IBM POWER9 support 16 MB or 2 MB
- PowerVM LPARs support 16 MB
- Bare-metal systems support 16 MB or 2 MB depending on your Linux kernel version

If they can get the settings from `/proc/meminfo `, should we just get them to check that instead and say something like:

- IBM Power Systems: Varies depending on kernel version, check `/proc/meminfo`

Reply:
@zl-wang - please see my last comment. Would you recommend getting the user to check `/proc/meminfo` in order to keep this content current going forward please? If not, is the list I printed ok?

Reply:
@SueChaplain I think you should go ahead with your suggested change until @zl-wang has further input.